modulemd/modulemd-buildopts.c

```diff
@@ -87,18 +93,20 @@ modulemd_buildopts_new (void)
 ModulemdBuildopts *
 modulemd_buildopts_copy (ModulemdBuildopts *self)
 {
-  g_autoptr (ModulemdBuildopts) b = NULL;
+  g_autoptr (ModulemdBuildopts) copy = NULL;
   g_return_val_if_fail (MODULEMD_IS_BUILDOPTS (self), NULL);
 
-  b = modulemd_buildopts_new ();
+  copy = modulemd_buildopts_new ();
 
-  modulemd_buildopts_set_rpm_macros (b,
+  modulemd_buildopts_set_rpm_macros (copy,
                                      modulemd_buildopts_get_rpm_macros (self));
 
-  g_hash_table_unref (b->whitelist);
-  b->whitelist = g_hash_table_ref (self->whitelist);
+  g_hash_table_unref (copy->whitelist);
+  copy->whitelist = g_hash_table_ref (self->whitelist);
```

Comment:
The original code here is buggy. We need to be making a deep copy of the hash table, not taking a reference on it. Otherwise, editing it in the new copy will modify it in the original copy (and vice-versa).

Mind fixing that for both whitelist and arches while you're here?

Reply:
Fixed.